devise-two-factor.gemspec

```
@@ -20,8 +20,8 @@ Gem::Specification.new do |s|
   s.test_files    = `git ls-files -- spec/*`.split("\n")
   s.require_paths = ['lib']
 
-  s.add_runtime_dependency 'railties',       '~> 7.0'
-  s.add_runtime_dependency 'activesupport',  '~> 7.0'
+  s.add_runtime_dependency 'railties',       '>= 7.0'
+  s.add_runtime_dependency 'activesupport',  '>= 7.0'
```

If the gem can be compatible with both Rails 7 and Rails 8 then I'd prefer to lock down the version to those two major versions specifically.
```suggestion
  s.add_runtime_dependency 'railties',       '~> 7.0, ~> 8.0'
  s.add_runtime_dependency 'activesupport',  '~> 7.0, ~> 8.0'
```

---

Does that format actually work? Seems like those things would be at odds with each other...

I'd expect either just a `>= 7.0` to be the change here, or also an addition of `< 8.1` if you want to be cautious on that end.

---

Yes, you're correct. I'd still like to keep an upper bound on the allowed versions but it would need to be `'>= 7.0'` and `'< 8.1'` or `'< 9'`. Looks like you've addressed that in #287.